File: apps/web/lib/types/briefings.ts
import type { DataMode } from "./status";

export type BriefingType =
  | "morning" | "client" | "legislative" | "crisis"
  | "media" | "geopolitical" | "sectorial";

export type BriefingAudience =
  | "consultor_politico" | "periodista" | "candidato"
  | "empresa_ibex" | "unidad_inteligencia" | "general";

export interface BriefingEvidence {
  id: string;
  title: string;
  source_id?: string | null;
  source_name?: string | null;
  url?: string | null;
  published_at?: string | null;
  excerpt?: string | null;
  confidence?: number | null;
  mode: DataMode;
}

export interface BriefingSection {
  id: string;
  type: string;
  title: string;
  body: string;
  bullets: string[];
  signals: string[];
  evidence: BriefingEvidence[];
  recommended_action?: string | null;
  target_route?: string | null;
  confidence?: number | null;
  mode: DataMode;
}

export interface BriefingRequest {
  briefing_type: BriefingType;
  audience: BriefingAudience;
  workspace_id?: string;
  client_id?: string | null;
  sector?: string | null;
  topic?: string | null;
  period?: string;
  force_refresh?: boolean;
  include_methodology?: boolean;
  include_evidence?: boolean;
  language?: "es" | "en";
}

export interface BriefingDocument {
  id: string;
  title: string;
  briefing_type: BriefingType;
  audience: BriefingAudience;
  workspace_id: string;
  client_id?: string | null;
  sector?: string | null;
  topic?: string | null;
  period: string;
  generated_at: string;
  mode: DataMode;
  model_used?: string | null;
  latency_ms?: number | null;
  executive_summary: string;
  sections: BriefingSection[];
  source_ids: string[];
  signal_ids: string[];
  warnings: string[];
  methodology_note?: string | null;
}

export interface BriefingListItem {
  id: string;
  title: string;
  briefing_type: BriefingType;
  audience: BriefingAudience;
  generated_at: string;
  mode: DataMode;
  workspace_id: string;
  client_id?: string | null;
  period: string;
  summary_preview: string;
}

export interface BriefingMarkdownResponse {
  mode: DataMode;
  briefing_id: string;
  markdown: string;
  filename: string;
}

export interface BriefingPdfResponse {
  mode: DataMode;
  briefing_id: string;
  format: "pdf" | "markdown";
  bytes_b64?: string;
  filename?: string;
  size?: number;
  markdown?: string;
  message?: string;
}

export interface BriefingsListResponse {
  mode: DataMode;
  items: BriefingListItem[];
  total: number;
}

File: api/schemas/briefings.py
"""
Briefing schemas for Sprint 3.
Reutiliza DataMode/ModeMeta de Sprint 1 en api/schemas/status.py.
"""
from __future__ import annotations

from datetime import datetime, timezone
from typing import Any, Literal
from pydantic import BaseModel, ConfigDict, Field

from api.schemas.status import DataMode, ModeMeta

BriefingType = Literal[
    "morning", "client", "legislative", "crisis", "media", "geopolitical", "sectorial",
]

BriefingAudience = Literal[
    "consultor_politico", "periodista", "candidato",
    "empresa_ibex", "unidad_inteligencia", "general",
]


class BriefingEvidence(BaseModel):
    id: str
    title: str
    source_id: str | None = None
    source_name: str | None = None
    url: str | None = None
    published_at: datetime | None = None
    excerpt: str | None = None
    confidence: float | None = None
    mode: DataMode = "real"


class BriefingSection(BaseModel):
    id: str
    type: str
    title: str
    body: str
    bullets: list[str] = Field(default_factory=list)
    signals: list[str] = Field(default_factory=list)
    evidence: list[BriefingEvidence] = Field(default_factory=list)
    recommended_action: str | None = None
    target_route: str | None = None
    confidence: float | None = None
    mode: DataMode = "real"


class BriefingRequest(BaseModel):
    briefing_type: BriefingType = "morning"
    audience: BriefingAudience = "general"
    workspace_id: str = "default"
    client_id: str | None = None
    sector: str | None = None
    topic: str | None = None
    period: str = "24h"
    force_refresh: bool = False
    include_methodology: bool = True
    include_evidence: bool = True
    language: Literal["es", "en"] = "es"


class BriefingDocument(BaseModel):
    id: str
    title: str
    briefing_type: BriefingType
    audience: BriefingAudience
    workspace_id: str = "default"
    client_id: str | None = None
    sector: str | None = None
    topic: str | None = None
    period: str = "24h"
    generated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    mode: DataMode = "real"
    model_used: str | None = None
    latency_ms: int | None = None
    executive_summary: str
    sections: list[BriefingSection] = Field(default_factory=list)
    source_ids: list[str] = Field(default_factory=list)
    signal_ids: list[str] = Field(default_factory=list)
    warnings: list[str] = Field(default_factory=list)
    methodology_note: str | None = None


class BriefingListItem(BaseModel):
    id: str
    title: str
    briefing_type: BriefingType
    audience: BriefingAudience
    generated_at: datetime
    mode: DataMode
    workspace_id: str
    client_id: str | None = None
    period: str = "24h"
    summary_preview: str
